cert/spec.go

```diff
@@ -50,9 +50,13 @@ type SpecOptions struct {
 	// This is Primarily useful to force an initial randomization if many nodes with certmgr are restarted all
 	// at the same time.
 	InitialSplay time.Duration
+
+	// KeyUsages specifies what this key will be used for, so that certmgr can verify it is valid for that usage.
+	// It is optional and by default we assume keys will be used for "TLS Web Server Authentication"
```

Comment:
Can you default elsewhere and make it required in this package?

Reply:
Ultimately this gets passed to golang's x509.Verify function which [defaults to ServerAuth](https://golang.org/src/crypto/x509/verify.go#L782) , so I think it's better to leave it optional throughout our codebase.